src/core/exec-invoke.c

```diff
@@ -2684,6 +2691,9 @@ static int compile_bind_mounts(
                         r = strv_consume(&empty_directories, private_root);
                         if (r < 0)
                                 return r;
+
+                        if (t == EXEC_DIRECTORY_STATE)
+                                idmapped = true;
```

Comment:
hmm, the !exec_context_with_rootfs(context) conditionalization looks wrong here?